### PR TITLE
{james wip}

### LINKED ISSUE #37: FR5 - Register for parties
| FR5: Register for parties |
|---|
| **Goal**: Users will be able to sign up for a party that they were either invited to or found through the UI |
| **Stakeholders**: Gamers / Party Hosts |
| This is one of the main features of the app. We need users to be able to sign-up for parties so that they can later meet up with others at the party. |
| **Origin**: This feature is essential and was thus thought up by Thadd. |
| **Version**: 1.0 |
| **Date**: 9/11/2020 |
| **Priority**: 1 |

### PR DESCRIPTION
**Pull request by codemonkeman**
Users can now register for parties. This will need some polish, but that's part of usability. It works. 
Closes #37

### CODE REVIEW

File: frontend/src/components/SearchUser/SearchUser.js [hunk 1]
```diff
@@ -1,23 +1,75 @@
...
-const { REACT_APP_URL } = process.env;
+  const { REACT_APP_URL } = process.env;
   const [user, setUser] = useContext(UserContext);
+  const [allUsers, setAllUsers] = useState([]);
   const [homeRender, setHomeRender] = useContext(HomeRenderContext);
-  const [parties, setParties] = useContext(PartiesContext);
+  const [search, setSearch] = useState('');
+
+
+  console.log('whatever in a string', allUsers);
```
Comment (ThaddG) [inline frontend/src/components/SearchUser/SearchUser.js:20]: That console.log isn't needed anymore.

File: frontend/src/components/SearchUser/SearchUser.js [hunk 2]
```diff
@@ -1,23 +1,75 @@
...
 import cookies from 'js-cookie';
 import './SearchUser.css';
 
 
 const SearchUser=(props)=>{
-const { REACT_APP_URL } = process.env;
+  const { REACT_APP_URL } = process.env;
   const [user, setUser] = useContext(UserContext);
+  const [allUsers, setAllUsers] = useState([]);
   const [homeRender, setHomeRender] = useContext(HomeRenderContext);
```
Comment (ThaddG) [inline frontend/src/components/SearchUser/SearchUser.js:16]: Don't need homeRender here. That's only for Home.js

File: frontend/src/components/SearchUser/SearchUser.js [hunk 3]
```diff
@@ -1,23 +1,75 @@
...
+      .catch((error) => console.log(error))
+  }
+
+  useEffect(() => {
+    getAllUsers();
+  }, [])
 
+  useEffect(()=>{
+    getAllUsers();
+  } , [homeRender])
```
Comment (ThaddG) [inline frontend/src/components/SearchUser/SearchUser.js:40]: This doesn't run, can be removed.

File: frontend/src/components/ViewParty/ViewParty.js [hunk 4]
```diff
@@ -10,6 +10,28 @@ import axios from 'axios';
...
 
+  const joinParty=() =>{
+
+    const headers = {
+      headers: {
+        "Content-Type": "application/json",
+      },
+    };
+  
+  const Link = `${process.env.REACT_APP_URL}/Party/${props.ID}/Update`;
```
Comment (ThaddG) [inline frontend/src/components/ViewParty/ViewParty.js]: That link has one too many '/'s. replace it with this:
const Link = `${process.env.REACT_APP_URL}Party/${props.ID}/Update`;

File: frontend/src/components/ViewParty/ViewParty.js [hunk 5]
```diff
@@ -10,6 +10,28 @@ import axios from 'axios';
...
+
+  const payload={
+    Attendees: props.Attendees.push({ID:props.user.ID,Username: props.user.Username})
```
Comment (ThaddG) [inline frontend/src/components/ViewParty/ViewParty.js]: I tested this out and it returns an integer. So to fix, declare an array above it, push/concat to it then let the value of the key be the declared array.

**Review by ThaddG — CHANGES_REQUESTED** [review]

**Comment by ga2599** [comment]
great work

**Pull request merged**